crates/router/src/core/payments/operations/payment_session.rs

```diff
@@ -70,9 +70,9 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsSessionRequest>
         )?;
 
         let mut payment_attempt = db
-            .find_payment_attempt_by_payment_id_merchant_id(
-                &payment_id,
+            .find_payment_attempt_by_merchant_id_attempt_id(
```

Comment:
we need payment_id also to fetch a payment_attempt. Without payment_id uniqueness of attempt can't be ensured. And for making the SQL and NoSql interfaces same we need payment_id also. And we might need indexes also for the same